src/compiler/checker.ts

```diff
@@ -14240,6 +14240,14 @@ namespace ts {
             return getObjectFlags(source) & ObjectFlags.JsxAttributes && !isUnhyphenatedJsxName(sourceProp.escapedName);
         }
 
+        function getNormalizedType(type: Type, writing: boolean): Type {

```

Comment:
How will this function jibe with https://github.com/microsoft/TypeScript/issues/31804 and the associated fix https://github.com/microsoft/TypeScript/pull/32116 - does this function just become recursive?

Reply:
We should be able to just move the loop into `getNormalizedType`.